MaterialLib/MPL/CreateComponent.cpp

```diff
@@ -72,28 +72,29 @@ std::vector<std::unique_ptr<Component>> createComponents(
     }
 
     std::vector<std::unique_ptr<Component>> components;
-
-    // Collect component's names to avoid duplicate components.
-    std::set<std::string> component_names;
-
     for (
         auto const& component_config :
         //! \ogs_file_param{prj__media__medium__phases__phase__components__component}
         config->getConfigSubtreeList("component"))
     {
         auto component = createComponent(component_config);
 
-        bool new_insertion = false;
-        std::tie(std::ignore, new_insertion) =
-            component_names.insert(component->name());
-        if (!new_insertion)
+        auto it = std::find_if(
```

Comment:
`BaseLib::findElementOrError()`

Reply:
At first glance, `BaseLib::findElementOrError()` is the first choice. This function returns custom error message if the element we are searching for is not found. It is opposite to what we want here - return error message if not found.

Reply:
Didn't see that! :fast_forward: